iss.py

```diff
@@ -1,11 +1,101 @@
 #!/usr/bin/env python
 
-__author__ = '???'
+__author__ = 'katran009 and google'
 
+import json
+import urllib.request
+import turtle
+import time
 
-def main():
-    pass
+# API addresses
+ASTRONAUTS_URL = 'http://api.open-notify.org/astros.json'
+LOCATION_URL = 'http://api.open-notify.org/iss-now.json'
 
+# Image paths
+BACKGROUND_PATH = 'images/map.gif'
+ISS_PATH = 'images/iss.gif'
 
-if __name__ == '__main__':
-    main()
+
+ISS = turtle.Turtle()
+WORLD = turtle.Screen()
+
+
+def api_data(url):
+    """
+    Get the JSON data from the specified NASA API.
+    """
+    url = url
+    response = urllib.request.urlopen(url)
+    # data came as bytes so convert into string to be parsed
+    string_response = response.read().decode("utf-8")
+    result = json.loads(string_response)
+
+    if result['message']:
+        return result
+    else:
+        return "Something went wrong."
+
+
+def display_craft_passengers():
+    """
+    Show number of astronauts and name of astronauts.
+    """
+    astros_data = api_data(ASTRONAUTS_URL)
+    number_astronauts = astros_data['number']
+    message = "There are " + str(number_astronauts) + \
+        " astronauts in space: \n"
+    # Save all astronauts plus a new line in a list
+    iss_passengers = [astronaut['name'] +
+                      "\n" for astronaut in astros_data['people']]
+
+    for person in iss_passengers:
+        message += person
```

Comment:
Instead of concatenating everything onto a base message, it may be easier to just print the number of astronauts as a header message, and iterate through the rest, like this
```python
# Part A: People in space and their spacecrafts
    astro_dict = get_astronauts()
    print('\nCurrent people in space: {}'.format(len(astro_dict)))
    for a in astro_dict:
        print(' - {} in {}'.format(a['name'], a['craft']))
```